server/modules/dataImport/service/DataImportJob.js

```diff
@@ -0,0 +1,152 @@
+import * as A from '@core/arena'
+
+import * as Survey from '@core/survey/survey'
+import * as NodeDef from '@core/survey/nodeDef'
+import * as Record from '@core/record/record'
+import * as Node from '@core/record/node'
+import { NodeValues } from '@core/record/nodeValues'
+
+import Job from '@server/job/job'
+import * as SurveyManager from '@server/modules/survey/manager/surveyManager'
+import * as RecordManager from '@server/modules/record/manager/recordManager'
+import { RecordsValidationBatchPersister } from '@server/modules/record/manager/RecordsValidationBatchPersister'
+import { NodesInsertBatchPersister } from '@server/modules/record/manager/NodesInsertBatchPersister'
+import { NodesUpdateBatchPersister } from '@server/modules/record/manager/NodesUpdateBatchPersister'
+
+import { DataImportFileReader } from './dataImportFileReader'
+
+export default class DataImportJob extends Job {
+  constructor(params) {
+    super(DataImportJob.type, params)
+  }
+
+  async execute() {
+    const { user, surveyId, tx } = this
+
+    this.nodesUpdateBatchPersister = new NodesUpdateBatchPersister({ user, surveyId, tx })
+    this.nodesInsertBatchPersister = new NodesInsertBatchPersister({ user, surveyId, tx })
+    this.recordsValidationBatchPersister = new RecordsValidationBatchPersister({ surveyId, tx })
+
```

Comment:
maybe add the tx as second param as the lien 42

Reply:
ok I'll change it in the next pr